pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord.go

```diff
@@ -275,3 +279,13 @@ func (c *dnsRecord) SetRecordType(recordType extensionsv1alpha1.DNSRecordType) {
 func (c *dnsRecord) SetValues(values []string) {
 	c.values.Values = values
 }
+
+func (c *dnsRecord) ValuesHaveChanged() bool {
+	return c.values.DNSName != c.dnsRecord.Spec.Name ||
+		!pointer.StringEqual(c.values.Zone, c.dnsRecord.Spec.Zone) ||
+		c.values.RecordType != c.dnsRecord.Spec.RecordType ||
+		!pointer.Int64Equal(c.values.TTL, c.dnsRecord.Spec.TTL) ||
+		!reflect.DeepEqual(c.values.Values, c.dnsRecord.Spec.Values) ||
+		c.secret.Name != c.dnsRecord.Spec.SecretRef.Name ||
```

Comment:
This is equivalent, but perhaps slightly better since it's more consistent with the other checks.
```suggestion
		c.values.SecretName != c.dnsRecord.Spec.SecretRef.Name ||
```

Reply:
done.